src/adapters/chineseloc/adapter.js

```diff
@@ -11,15 +11,15 @@ class AlpheiosChineseLocAdapter extends BaseAdapter {
 
   get languageID () { return ChineseLanguageModel.languageID }
 
-  fetchChineseData (targetWord) {
+  fetchChineseData (targetWord, checkContextForward) {
     ChineseSource.collectData()
 
-    return ChineseSource.lookupChinese(targetWord)
+    return ChineseSource.lookupChinese(targetWord, checkContextForward)
   }
 
-  getHomonym (targetWord) {
+  getHomonym (targetWord, checkContextForward) {
     try {
-      const res = this.fetchChineseData(targetWord)
+      const res = this.fetchChineseData(targetWord, checkContextForward)

```

Comment:
there is something not quite right here. Take the following use case from the issue:

with this text

安之若素的意

we find 4 results

0: {word: "安 安", pinyin: "an", shortDef: "content; calm; still; quiet; to pacify; peace", format: "simplified", dictEntry: "安", …}
1: {word: "安 安", pinyin: "an", shortDef: "to pacify", format: "simplified", dictEntry: "安", …}
2: {word: "安之若素 安之若素", pinyin: "an zhi ruò sù", shortDef: "bear hardship with equanimity; regard wrongdoing with equanimity", format: "simplified", dictEntry: "安之若素", …}
3: {word: "安之若素 安之若素", pinyin: "an zhi ruò sù", shortDef: "bear hardship with equanimity", format: "simplified", dictEntry: "安之若素", …}
length: 4

Right now these are getting converted to a single homonym, with 4 lexemes. But actually we have two distinct homonyms here:

安

and 

安之若素

Each of these (apparently) have 2 lexemes. 

So I think when we create the homonyn response, we need to group the results by distinct word, and have each of those turn into distinct homonyms. I'm not sure what the implications are of this for the components code though ... I think we might need to make additional changes there to support multiple homonyms in the popup.

Reply:
that might be more than we can manage this week though...

Reply:
it is not a problem to rearrange one Adapter - to return several homonyms.
But components repo - LexicalQuery, Vuex store, Vue-components, WordListController  are not ready to multiple homonyms at a time.
@balmas, @kirlat, May be we should discuss such an upgrade first?

Reply:
I agree that we need to discuss it because it might have implications across our whole architecture. Maybe you should open an issue for that? There we can discuss what parts of an application are affected and how would it be best to deal with the changes.

Reply:
Then I sugest the following

I created a separate PR (https://github.com/alpheios-project/client-adapters/pull/43) for only this issue - https://github.com/alpheios-project/client-adapters/issues/41

And I created a specific issue for multiple homonyms - 
https://github.com/alpheios-project/components/issues/856

And I would create a specific PR later for ClientAdapters - thet would return Homonym array or HomonymSet from Chinese Adapter (according to the chosen solution in the issue discussion)

For now I will close this PR